trader/trader.go

```diff
@@ -615,30 +616,29 @@ func (t *Trader) setExistingOffers(sellingAOffers []hProtocol.Offer, buyingAOffe
 	t.sellingAOffers, t.buyingAOffers = sellingAOffers, buyingAOffers
 }
 
-func (t *Trader) countOfferChangeTypes(offers []txnbuild.Operation) (numCreate int, numUpdate int, numDelete int, e error) {
-	var offerXdr xdr.Operation
+func countOfferChangeTypes(offers []*txnbuild.ManageSellOffer) (int, int, int, error) {
+	numCreate, numDelete, numUpdate := 0, 0, 0
 	for _, o := range offers {
-		offerXdr, e = o.BuildXDR()
-		if e != nil {
-			return 0, 0, 0, fmt.Errorf("could not convert op to offer: %s", e)
+		if o == nil {
+			return 0, 0, 0, fmt.Errorf("offers were not of expected type ManageSellOffer")
```

Comment:
we should provide more context on the failure so it's easier to debug. such as:
`fmt.Errorf("offer at index %d was not of expected type ManageSellOffer (actual type = %T): %+v", i, o, o)`